src/test/server/store-test.js

```diff
@@ -0,0 +1,201 @@
+    website: false
+};
+
+let users;
+let foundUsers;
+let searchFields;
+let userId;
+
+UserStore.on(EVENT_TYPE.usersFound, _getFoundUsers);
+UserStore.on(EVENT_TYPE.addingFailed, _getUserID);
+
+function _getFoundUsers (users, fields) {
+    foundUsers   = users;
+    searchFields = fields;
+}
+
+function _getUserID (id) {
+    userId = id;
+}
+
+function _initUserList () {
+    return usersData.map((user) => {
+        return new UserInfo(
+            {
+                id:      user.id,
+                name:    user.name,
+                phone:   user.phone,
+                email:   user.email,
+                website: user.website
+            },
+            user.address,
+            user.company
+        );
+    });
+}
+
+describe('UserStore', () => {
+    beforeEach(() => {
+        users = _initUserList();
+    });
+
+    afterEach(() => {
+        UserStore._users        = _initUserList();
+        UserStore._searchedUser = {};
+    });
+
+    describe('.getUsers()', () => {
+        it('Should return a list of users', () => {
+            assert(UserStore.getUsers(), users);
+        });
+    });
+
+    describe('Actions', () => {
+        describe('addNewUser', () => {
+            it('Should emit userAdded event', () => {
+                users.push(TEST_USER);
+                UserAction.addNewUser(TEST_USER);
+                assert(UserStore.getUsers(), users);
+                assert(UserStore._users[UserStore._users.length - 1].id, UserStore._users.length);
+            });
+
+            it('Should emit addingFailed event with duplicateUserId if user exists ', () => {
+                UserAction.addNewUser(TEST_USER_EXISTING);
+                assert(UserStore.getUsers(), users);
+                assert(userId, 2);
+            });
+
+            it('Should emit userAdded event if user exists, but force flag is true', () => {
+                users.push(TEST_USER_EXISTING);
+                UserAction.addNewUser(TEST_USER_EXISTING, true);
+                assert(UserStore.getUsers(), users);
+            });
+
+            it('Should emit usersFound event with foundUsers and searchFields if we are in search mode', () => {
+                users.push(TEST_USER);
+                UserAction.findUser(SEARCH_USER_INFO);
+                UserAction.addNewUser(TEST_USER);
+                assert(UserStore.getUsers(), users);
+                assert(foundUsers, []);
+                assert(searchFields, SEARCH_FIELDS);
+            });
+        });
+
+        describe('updateUser', () => {
+            it('Should emit change event', () => {
+                assert.notEqual(users[1], UPDATE_USER);
+                users[1] = UPDATE_USER;
+                UserAction.updateUser(UPDATE_USER);
+                assert(UserStore.getUsers(), users);
+            });
+
+            it('Should emit usersFound event with foundUsers and searchFields if update in search mode', () => {
+                UserAction.findUser(SEARCH_USER_INFO);
```

Comment:
(proposal)
we can set `this._searchedUser` instead of execute real `findUser` (we chech that searching is correct in the corresponding tests, we can skip checking it here; we need just simulate searching mode)